[KimHan01] iP

KimHan01 opened the pull request:
# Rocket 🦝 

Are you forgetful? Get a wake-up call with Rocket. It's,
- text-based
- simple to learn
- ~~FAST~~ LIGHTNING FAST to use

All you need to do is,
1. download it from [here](https://github.com/KimHan01/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE**

Features:
- [x] Manage tasks
- [x] Manage deadlines
- [x] Manage events 
#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) {
        String filePath = "./data/storage.txt";
        new Rocket(filePath).run();
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

KimHan01 committed:
Replace Duke.java with Rocket.java

KimHan01 committed:
Rocket.java: Add echo functionality to Rocket.

KimHan01 committed:
Rocket.java: Add functionality to add input to list with responses to some invalid actions

KimHan01 committed:
Move Rocket.java into the package chatbot. Create Response class inside package.

KimHan01 committed:
Create class for Task. Add functionality to mark tasks as done and not done

KimHan01 committed:
Add Todo, Deadline and Event.

KimHan01 committed:
Remove taskNum from Task.
runtest.bat: updated javac and java to match location of main class

KimHan01 committed:
Updated input.txt with inputs and expected.txt with expected outputs

KimHan01 committed:
Add Error handling for empty task and invalid input

KimHan01 committed:
Add support for deleting tasks from the list.

KimHan01 committed:
Rocket.java: Add code for creating a storage text file

KimHan01 committed:
Move code for creating storage file into dataHandler class

KimHan01 committed:
Rename storage.txt to list.txt
Task.java: Add getter for task name
dataHandler: Move dir and file out of createListFile() into static variables of dataHandler, and add code for readList and addTodo
Rocket.java: Add code to call dataHandler into read list and add todo

KimHan01 committed:
Ignore list.txt

KimHan01 committed:
dataHandler: add todo, deadline, event

KimHan01 committed:
dataHandler: Combine addTask, addDeadline, addEvent into addTask

KimHan01 committed:
add function to delete task from txt file

KimHan01 committed:
add functions to mark task and unmark task in txt file

KimHan01 committed:
dataHandler: remove redundant markTask, unmarkTask and deleteTask. Instead, add updateList

KimHan01 committed:
Merge branch 'branch-Level-7'

KimHan01 committed:
Add CustomDateFormatter.java to format dates

KimHan01 committed:
Merge branch 'branch-Level-8'

KimHan01 committed:
Major refactor to follow MoreOOP style

KimHan01 committed:
Move classes into packages, and fixed a minor bug of tasks not being added to storage.txt

KimHan01 committed:
Merge branch 'branch-A-Packages'

KimHan01 committed:
Merge branch 'add-gradle-support'

KimHan01 committed:
correct mainclass in build.gradle

KimHan01 committed:
Add deadlineTest and TodoTest

KimHan01 committed:
Remove some print messages

KimHan01 committed:
change filePath in main

KimHan01 committed:
add functionality to find tasks by keyword in the task name

KimHan01 committed:
Merge branch 'branch-Level-9'

DanielJames0302 commented on src/main/java/rocket/command/Command.java:
```diff
@@ -0,0 +1,19 @@
+        this.isExit = isExit;
+    }
+
+    public boolean isExit() {
```
Good method naming

DanielJames0302 commented on src/main/java/rocket/command/Command.java:
```diff
@@ -0,0 +1,19 @@
+import rocket.ui.Ui;
+
+public abstract class Command {
+    private final boolean isExit; // Checks the command is an exit command
```
Good boolean variable naming

DanielJames0302 commented on src/main/java/rocket/parser/Parser.java:
```diff
@@ -0,0 +1,221 @@
+    public static Command parse(String input) {
+        String commandType = getCommandType(input);
+        switch (commandType) {
+            case TODO:
```
Should the `case` not be indented ?

DanielJames0302 reviewed (commented):
LGTM. Just some nits to fix

Ernestnzx commented on src/main/java/rocket/command/ListCommand.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public void execute(TaskList list, Ui ui, Storage storage) {
+        if (list.getSize() == 0) {
```
Do consider `list.isEmpty()` method for ease of readability.

Ernestnzx commented on src/main/java/rocket/command/FindCommand.java:
```diff
@@ -0,0 +1,26 @@
+import rocket.ui.Ui;
+
+public class FindCommand extends Command {
+    String keyword;
```
Small nit, can change the visibility of the variable to be `private final`.

Ernestnzx commented on src/main/java/rocket/parser/Parser.java:
```diff
@@ -0,0 +1,221 @@
+package rocket.parser;
+
+import rocket.formatter.CustomDateFormatter;
+import rocket.command.*;
```
Wildcard imports are not allowed. Explicitly declare the classes that you are going to use.

Ernestnzx commented on src/main/java/rocket/storage/Storage.java:
```diff
@@ -0,0 +1,120 @@
+package rocket.storage;
+
+import rocket.task.*;
```
Wildcard imports are not allowed, explicitly declare the classes that you are going to use.

Ernestnzx reviewed (commented):
LGTM, apart from a few nits the code is well written and understandable. 

Keep up the good work! ✨

zyonwee commented on src/main/java/rocket/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+import rocket.formatter.CustomDateFormatter;
+
+import java.time.LocalDate;
+
```
Perhaps you could write descriptive header comments for all public classes?

zyonwee commented on src/main/java/rocket/task/TaskList.java:
```diff
@@ -0,0 +1,67 @@
+                LocalDate to = task.getTo();
+                return new Event(name, mark, from, to);
+            default:
+                throw new RocketRuntimeException("Failed to get Task from TaskList");
```
I think you may want to remove case indents and add a fall through comment since break was ommited from your switch case as per our briefing notes.

```suggestion
        switch (taskType) {
        case TODO:
            return new Todo(name, mark);
        case DEADLINE:
            LocalDate by = task.getBy();
            return new Deadline(name, mark, by);
        case EVENT:
            LocalDate from = task.getFrom();
            LocalDate to = task.getTo();
            return new Event(name, mark, from, to);
        default:
            throw new RocketRuntimeException("Failed to get Task from TaskList");
            // Add Fall Through Comment
```

zyonwee commented on src/main/java/rocket/ui/Ui.java:
I like your implementation of a UI file separately

zyonwee commented on src/main/java/rocket/common/Utils.java:
I like your implementation of a Utility file

zyonwee commented on src/main/java/rocket/storage/Storage.java:
```diff
@@ -0,0 +1,120 @@
+        Scanner sc = new Scanner(file);
+        while (sc.hasNext()) {
+            String[] parts = sc.nextLine().split("\\|", 2);
+            String header = parts[0];
+            String body = parts[1];
+            try {
+                switch (header) {
+                    case "T":
+                        Todo todo = Todo.fromTxt(body);
+                        tasks.add(todo);
+                        break;
+                    case "D":
+                        Deadline deadline = Deadline.fromTxt(body);
+                        tasks.add(deadline);
+                        break;
+                    case "E":
+                        Event event = Event.fromTxt(body);
+                        tasks.add(event);
+                        break;
+                }
```
You may consider removing the indentations for case

zyonwee reviewed (commented):
LGTM. Just a few nits to fix

KimHan01 committed:
Rocket.java: Add javadocs

KimHan01 committed:
Ui.java: Add javadocs

KimHan01 committed:
Add javadocs to all classes under task directory

KimHan01 committed:
Update javadocs in deadline and event class

KimHan01 committed:
add javadocs to all command classes

KimHan01 committed:
add javadocs to utils, exceptions, parser and storage classes

KimHan01 committed:
Update code to adhere to coding standards

KimHan01 committed:
Merge branch 'branch-A-JavaDoc'

KimHan01 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/rocket/command/EmptyTaskNameCommand.java
#	src/main/java/rocket/parser/Parser.java

KimHan01 committed:
get Javafx integrated and GUI to work

KimHan01 committed:
Get rocket to respond to user chat message in GUI

KimHan01 committed:
Add javafx stage title and icon

KimHan01 committed:
Merge branch 'branch-Level-10'

KimHan01 committed:
Add help command

KimHan01 committed:
Remove unneccesary methods in Ui.java

KimHan01 committed:
build.gradle: Change main class to Launcher

KimHan01 committed:
Storage.java: Add asserts to check filepath

KimHan01 committed:
Improve code quality for entire codebase

KimHan01 committed:
Ui.java: slight reformat

KimHan01 committed:
Deadline.java: removed an unneccesary comment

KimHan01 committed:
Slight reformat to event and ui class

KimHan01 committed:
Add asserts to Parser and Storage class.

There are methods in Parser and Storage class which attempts to
cut the string at a certain point to obtain information.

Asserts are added inside these methods right before cutting

This ensures programmer can test whether these methods are given
the string of sufficient length before cutting.

KimHan01 committed:
Merge branch 'branch-A-Assertions' of https://github.com/KimHan01/ip into branch-A-Assertions

KimHan01 committed:
Remove some unnecessary asserts

KimHan01 committed:
Merge pull request from KimHan01/branch-A-Assertions

Branch-A-Assertions PR

KimHan01 committed:
Merge master branch into branch-A-CodeQuality

branch-A-CodeQuality is no longer in sync with latest master branch.

Need to merge master branch into branch-A-CodeQuality to sync
changes from latest master branch.

KimHan01 committed:
Merge pull request from KimHan01/branch-A-CodeQuality

Branch-A-Code Quality PR

KimHan01 committed:
Slight format update to CustomDateFormatter.java

KimHan01 committed:
Refactor Command classes

KimHan01 committed:
Add edit functionality

KimHan01 committed:
Change rocket's intro to be more concise and did slight refactor for EditDateCommand and HelpCommand

KimHan01 committed:
Merge branch 'branch-update'

KimHan01 committed:
Add javaDocs for edit command classes, Change some responses

KimHan01 committed:
Update README, add Ui image and update help command

KimHan01 committed:
Update GUI to look better, add checkstyle, and add delay before exit

KimHan01 committed:
Change location of checkstyle files

KimHan01 committed:
Merge branch 'branch-improveGUI'

KimHan01 committed:
MainWindow.java: Slight rename of a single line

KimHan01 committed:
Update edit index out of bounds message

KimHan01 committed:
Update farewell message

KimHan01 committed:
Update invalid unmark response

KimHan01 committed:
Update invalid format response

KimHan01 committed:
Update README with delete feature

KimHan01 committed:
test
